src/middlewared/middlewared/plugins/disk_/zfs_guid.py

```diff
@@ -118,6 +118,16 @@ async def zfs_events_hook(middleware, data):
         except MatchNotFound:
             pass
 
+        # We do this separately and do not get it from above because right now above implementation would handle
```

Comment:
The code has nothing to do with ZFS GUIDs :) Let's move this to a separate file and a separate event handler